src/tigerbeetle/scrub.zig

```diff
@@ -0,0 +1,510 @@
+const std = @import("std");
+const stdx = vsr.stdx;
+const assert = std.debug.assert;
+const log = std.log.scoped(.scrub);
+
+const cli = @import("cli.zig");
+const vsr = @import("vsr");
+const constants = vsr.constants;
+const IO = vsr.io.IO;
+const StateMachine = @import("main.zig").StateMachine;
+const Replica = @import("main.zig").Replica;
+const Storage = @import("main.zig").Storage;
+const Grid = @import("main.zig").Grid;
+const Tracer = vsr.trace.Tracer;
+
+const Forest = StateMachine.Forest;
+const CheckpointTrailer = vsr.CheckpointTrailerType(Storage);
+const SuperBlock = Replica.SuperBlock;
+
+/// Special offline GridScrubber type - uses constants.grid_iops_read_max rather than
+/// constants.grid_scrubber_reads_max.
+const GridScrubber = vsr.GridScrubberType(Forest, constants.grid_iops_read_max);
+
+pub fn command_scrub(
+    gpa: std.mem.Allocator,
+    io: *IO,
+    tracer: *Tracer,
+    args: *const cli.Command.Scrub,
+) !void {
+    var scrub: Scrub = undefined;
+    var scrubbed_bytes: u64 = 0;
+
+    // The superblock is checked as part of initalizing the scrubber - it cannot be skipped.
+    const scrubbed_bytes_superblock = try Scrub.init(
```

Comment:
Returning this from `Scrub.init` seems awkward to me... I think we could just add it as

```
    for (scrub.superblock.reading) |_| {
        scrubbed_bytes += vsr.superblock.superblock_copy_size;
    }
```

here in `command_scrub`. Or alternatively, we don't consider this "scrubbing" (since we don't check all copies anyway) and instead just say that an openable superblock is a prerequisite for scrubbing, and an unopenable superblock is a fatal error.

Reply:
Yeah, it is a little awkward, but I really wanted to keep the scrubbed bytes calculation both split (eg, the actual and the target are calculated separately) and local (immediately after checking said blocks, increment the counter).

I think I'm going to make `open()` also assert the superblock is open and verified, and it can return the bytes checked.